### PR TITLE
Add table aws_wafv2_regex_pattern_set. closes #272

### LINKED ISSUE #272: Add table aws_wafv2_regex_pattern_set
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go/service/wafv2

### PR DESCRIPTION
**Pull request by sayan133**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_wafv2_regex_pattern_set []

PRETEST: tests/aws_wafv2_regex_pattern_set

TEST: tests/aws_wafv2_regex_pattern_set
Running terraform
aws_wafv2_regex_pattern_set.named_test_resource_regional: Creating...
aws_wafv2_regex_pattern_set.named_test_resource_global: Creating...
aws_wafv2_regex_pattern_set.named_test_resource_global: Creation complete after 4s [id=4d32ef26-6742-4229-b2bb-6c197e8106de]
aws_wafv2_regex_pattern_set.named_test_resource_regional: Creation complete after 6s [id=e95e3e61-f609-42f6-909c-9973ff38a4ad]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

account_id = "388460667113"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka_global = "arn:aws:wafv2:us-east-1:388460667113:global/regexpatternset/turbottest82310_global/4d32ef26-6742-4229-b2bb-6c197e8106de"
resource_aka_regional = "arn:aws:wafv2:us-east-1:388460667113:regional/regexpatternset/turbottest82310_regional/e95e3e61-f609-42f6-909c-9973ff38a4ad"
resource_id_global = "4d32ef26-6742-4229-b2bb-6c197e8106de"
resource_id_regional = "e95e3e61-f609-42f6-909c-9973ff38a4ad"
resource_name_global = "turbottest82310_global"
resource_name_regional = "turbottest82310_regional"

Running SQL query: test-getGlobal-query.sql
[
  {
    "account_id": "388460667113",
    "arn": "arn:aws:wafv2:us-east-1:388460667113:global/regexpatternset/turbottest82310_global/4d32ef26-6742-4229-b2bb-6c197e8106de",
    "description": "A global Regex Pattern set for testing.",
    "id": "4d32ef26-6742-4229-b2bb-6c197e8106de",
    "name": "turbottest82310_global",
    "partition": "aws",
    "region": "global",
    "regular_expressions": [
      "^turbot",
      "^google"
    ],
    "scope": "CLOUDFRONT"
  }
]
✔ PASSED

Running SQL query: test-getRegional-query.sql
[
  {
    "account_id": "388460667113",
    "arn": "arn:aws:wafv2:us-east-1:388460667113:regional/regexpatternset/turbottest82310_regional/e95e3e61-f609-42f6-909c-9973ff38a4ad",
    "description": "A regional Regex Pattern set for testing.",
    "id": "e95e3e61-f609-42f6-909c-9973ff38a4ad",
    "name": "turbottest82310_regional",
    "partition": "aws",
    "region": "us-east-1",
    "regular_expressions": [
      "^turbot",
      "^google"
    ],
    "scope": "REGIONAL"
  }
]
✔ PASSED

Running SQL query: test-hydrateGlobal-query.sql
[
  {
    "description": "A global Regex Pattern set for testing.",
    "name": "turbottest82310_global",
    "tags_src": [
      {
        "Key": "scope",
        "Value": "global"
      },
      {
        "Key": "foo1",
        "Value": "bar1"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-hydrateRegional-query.sql
[
  {
    "description": "A regional Regex Pattern set for testing.",
    "name": "turbottest82310_regional",
    "tags_src": [
      {
        "Key": "foo",
        "Value": "bar"
      },
      {
        "Key": "scope",
        "Value": "regional"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-listGlobal-query.sql
[
  {
    "arn": "arn:aws:wafv2:us-east-1:388460667113:global/regexpatternset/turbottest82310_global/4d32ef26-6742-4229-b2bb-6c197e8106de",
    "id": "4d32ef26-6742-4229-b2bb-6c197e8106de",
    "name": "turbottest82310_global",
    "region": "global",
    "scope": "CLOUDFRONT"
  }
]
✔ PASSED

Running SQL query: test-listRegional-query.sql
[
  {
    "arn": "arn:aws:wafv2:us-east-1:388460667113:regional/regexpatternset/turbottest82310_regional/e95e3e61-f609-42f6-909c-9973ff38a4ad",
    "id": "e95e3e61-f609-42f6-909c-9973ff38a4ad",
    "name": "turbottest82310_regional",
    "region": "us-east-1",
    "scope": "REGIONAL"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:wafv2:us-east-1:388460667113:regional/regexpatternset/turbottest82310_regional/e95e3e61-f609-42f6-909c-9973ff38a4ad"
    ],
    "tags": {
      "foo": "bar",
      "scope": "regional"
    },
    "title": "turbottest82310_regional"
  }
]
✔ PASSED

POSTTEST: tests/aws_wafv2_regex_pattern_set

TEARDOWN: tests/aws_wafv2_regex_pattern_set

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
### Basic info

```sql
select
  name,
  description,
  arn,
  id,
  scope,
  regular_expressions,
  region
from
  aws_wafv2_regex_pattern_set;
```

```
+-------------------------+--------------------------------------+----------------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+---------------------------+-----------+
| name                    | description                          | arn                                                                                                                        | id                                   | scope      | regular_expressions       | region    |
+-------------------------+--------------------------------------+----------------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+---------------------------+-----------+
| test-steampipe-regional | Regional Regex Pattern Set for test  | arn:aws:wafv2:us-east-1:388460667113:regional/regexpatternset/test-steampipe-regional/ac20dc94-65ff-4a75-ae1c-21e9664520bc | ac20dc94-65ff-4a75-ae1c-21e9664520bc | REGIONAL   | ["^turbot","^www.google"] | us-east-1 |
| test-steampipe-global   | Global Regex Pattern Set for testing | arn:aws:wafv2:us-east-1:388460667113:global/regexpatternset/test-steampipe-global/92dc5a85-a51c-4439-81f8-c8fdf3d5a0fc     | 92dc5a85-a51c-4439-81f8-c8fdf3d5a0fc | CLOUDFRONT | ["^www.turbot"]           | global    |
+-------------------------+--------------------------------------+----------------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+---------------------------+-----------+

```

### List global (CLOUDFRONT) Regex Pattern sets

```sql
select
  name,
  description,
  arn,
  id,
  scope,
  regular_expressions,
  region
from
  aws_wafv2_regex_pattern_set
where
  scope = 'CLOUDFRONT';
```

```
+-----------------------+--------------------------------------+------------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+---------------------+--------+
| name                  | description                          | arn                                                                                                                    | id                                   | scope      | regular_expressions | region |
+-----------------------+--------------------------------------+------------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+---------------------+--------+
| test-steampipe-global | Global Regex Pattern Set for testing | arn:aws:wafv2:us-east-1:388460667113:global/regexpatternset/test-steampipe-global/92dc5a85-a51c-4439-81f8-c8fdf3d5a0fc | 92dc5a85-a51c-4439-81f8-c8fdf3d5a0fc | CLOUDFRONT | ["^www.turbot"]     | global |
+-----------------------+--------------------------------------+------------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+---------------------+--------+

```

### List Regex Pattern sets having a specific regex pattern

```sql
select
  name,
  description,
  arn,
  id,
  scope,
  regular_expressions,
  region
from
  aws_wafv2_regex_pattern_set,
  jsonb_array_elements_text(regular_expressions) as regex
where
  regex = '^turbot';
```

```
+-------------------------+-------------------------------------+----------------------------------------------------------------------------------------------------------------------------+--------------------------------------+----------+---------------------------+-----------+
| name                    | description                         | arn                                                                                                                        | id                                   | scope    | regular_expressions       | region    |
+-------------------------+-------------------------------------+----------------------------------------------------------------------------------------------------------------------------+--------------------------------------+----------+---------------------------+-----------+
| test-steampipe-regional | Regional Regex Pattern Set for test | arn:aws:wafv2:us-east-1:388460667113:regional/regexpatternset/test-steampipe-regional/ac20dc94-65ff-4a75-ae1c-21e9664520bc | ac20dc94-65ff-4a75-ae1c-21e9664520bc | REGIONAL | ["^turbot","^www.google"] | us-east-1 |
+-------------------------+-------------------------------------+----------------------------------------------------------------------------------------------------------------------------+--------------------------------------+----------+---------------------------+-----------+

```
</details>

### CODE REVIEW

File: aws/table_aws_wafv2_regex_pattern_set.go [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+				Hydrate:     getCommonColumns,
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsWafv2RegexPatternSets(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_wafv2_regex_pattern_set.go:123]:
```suggestion
```

File: aws/table_aws_wafv2_regex_pattern_set.go [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsWafv2RegexPatternSet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAwsWafv2RegexPatternSet")
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_wafv2_regex_pattern_set.go:174]:
```suggestion
```

File: aws/table_aws_wafv2_regex_pattern_set.go [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+			{
+				Name:        "description",
+				Description: "A description of the Regex Pattern set that helps with identification.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "lock_token",
+				Description: "A token used for optimistic locking.",
+				Type:        proto.ColumnType_STRING,
+			},
```
Comment (khushboo9024) [inline aws/table_aws_wafv2_regex_pattern_set.go:60]: Are we getting lock_token in get call , please check once.

File: aws/table_aws_wafv2_regex_pattern_set.go [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+		turbotTagsMap = map[string]string{}
+		for _, i := range data.TagInfoForResource.TagList {
+			turbotTagsMap[*i.Key] = *i.Value
+		}
+	}
+
+	return turbotTagsMap, nil
+}
+
+func RegularExpressionObjectListToRegularExpressionList(ctx context.Context, d *transform.TransformData) (interface{}, error) {
```
Comment (Subhajit97) [inline aws/table_aws_wafv2_regex_pattern_set.go:309]: Use lowerCamelCase when defining any functions

File: aws/table_aws_wafv2_regex_pattern_set.go [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+	op, err := svc.GetRegexPatternSet(params)
+	if err != nil {
+		plugin.Logger(ctx).Debug("GetRegexPatternSet", "ERROR", err)
+		return nil, err
+	}
+
+	return op.RegexPatternSet, nil
+}
+
+func getAwsWafv2RegexPatternSetLockToken(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table_aws_wafv2_regex_pattern_set.go:240]: We can avoid using get api call twice rather we can use new version of `FromField` where we can pass more than one path in comma separated  form.

Comment (bigdatasourav) [reply aws/table_aws_wafv2_regex_pattern_set.go]: updated

**Review by khushboo9024 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**